### PR TITLE
Create cli-runtime staging repository

### PR DESCRIPTION
**Pull request by soltysh**
**What this PR does / why we need it**:
This PR creates a designated repository for CLI helpers, which are used for:
- kubectl itself
- kubectl plugins
- commands that want to mimic kubectl behavior

/assign @deads2k @juanvallejo 
@seans3 @pwittrock fyi

**Release note**:
```release-note
Create cli-runtime staging repository
```

### CODE REVIEW

File: staging/src/k8s.io/cli-runtime/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+`cli-runtime` is synced from https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cli-runtime.
+Code changes are made in that location, merged into `k8s.io/kubernetes` and later synced here.
+
+
+## Things you should *NOT* do
+
+ 1. Add API types to this repo.  This is for the helpers, not for the types.
+ 2. Directly modify any files under `pkg` in this repo.  Those are driven from `k8s.io/kubernetes/staging/src/k8s.io/cli-runtime`.
+ 3. Expect compatibility.  This repo is direct support of Kubernetes and the API isn't yet stable enough for API guarantees.
```
Comment (deads2k) [inline staging/src/k8s.io/cli-runtime/README.md]: perhaps 4.  Add any type that only makes sense for `kubectl`.  This isn't a helper for `kubectl`, this is a helper for `kubectl`-like command line tools. Think things that make cli plugins easier to write.

**Comment by deads2k** [comment]
lgtm

Get @sttts to have a look and see if there's anything else that needs doing before creating this.

**Comment by sttts** [comment]
Seems good to me. A counterpart PR like https://github.com/kubernetes/publishing-bot/pull/89 would be nice.

**Comment by sttts** [comment]
/assign @smarterclayton 

for approval of the new staging dir.

File: staging/src/k8s.io/cli-runtime/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# cli-runtime
+
+Set of helpers for creating kubectl plugins as well as kubectl compatible commands.
```
Comment (juanvallejo) [inline staging/src/k8s.io/cli-runtime/README.md]: Given that we are planning on dog-fooding this package ourselves in kubectl, maybe: `Set of helpers for creating kubectl commands, as well as kubectl plugins`?

**Comment by soltysh** [comment]
I've fixed the comments from David and Juan wrt Readme.

**Comment by smarterclayton** [comment]
Can you explain a bit the naming of the repo?  I.e. why `cli-runtime` vs other names?  Or link to that discussion?

**Comment by soltysh** [comment]
> Can you explain a bit the naming of the repo? I.e. why cli-runtime vs other names? Or link to that discussion?

@smarterclayton @pwittrock and I picked that name, based on the previous repos that are available like controller-runtime, with similar functionality. This was also presented during our previous sig-cli meeting everyone agreed with the name. If you have a better proposal I'm open.

**Comment by neolit123** [comment]
/sig cli
for the release note tool.

**Pull request merged**